lib/galaxy/objectstore/cloud.py

```diff
@@ -0,0 +1,615 @@
+"""
+Object Store plugin for Cloud storage.
+"""
+
+import logging
+import multiprocessing
+import os
+import shutil
+import subprocess
+import threading
+import time
+
+from datetime import datetime
+
+from galaxy.exceptions import ObjectInvalid, ObjectNotFound
+from galaxy.util import (
+    directory_hash_id,
+    safe_relpath,
+    string_as_bool,
+    umask_fix_perms,
+)
+from galaxy.util.sleeper import Sleeper
+
+from ..objectstore import convert_bytes, ObjectStore
+from cloudbridge.cloud.factory import CloudProviderFactory, ProviderList
+
+try:
+    # Imports are done this way to allow objectstore code to be used outside of Galaxy.
+    import boto
+
+    from boto.exception import S3ResponseError
+    from boto.s3.key import Key
```

Comment:
The linter doesn't like unused imports (this line and the following), but you can ignore this by doing
```
from boto.s3.key import Key  # noqa: F401
from boto.s3.connection import S3Connection  # noqa: F401
```

Although I'm not sure in which scenario it would help to import unused dependencies ...
If external code needs it, it should probably import the dependencies where they are needed.